ccl/multiregionccl: TestMultiRegionDataDriven_global_tables failed

ccl/multiregionccl.TestMultiRegionDataDriven_global_tables [failed](https://mesolite.cluster.engflow.com/invocations/default/6e1a17bc-56f5-4387-886a-6c6b3885dc67?testReportRun=4&testReportShard=2&testReportAttempt=1#targets-Ly9wa2cvY2NsL211bHRpcmVnaW9uY2NsOm11bHRpcmVnaW9uY2NsX3Rlc3Q=) on master @ [f1b136a030d439e5dc89a27cb4de08045dca8e77](https://github.com/cockroachdb/cockroach/commits/f1b136a030d439e5dc89a27cb4de08045dca8e77):


```
          new-cluster localities=us-east-1,us-east-1,us-east-1,us-west-1,us-central-1,eu-central-1
          ----
    datadriven_test.go:151: 
        /mnt/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/global_tables:4:
          exec-sql idx=0
          CREATE DATABASE db PRIMARY REGION "us-east-1" REGIONS  "us-west-1", "us-central-1";
          ----
    datadriven_test.go:151: 
        /mnt/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/global_tables:8:
          exec-sql idx=0
          CREATE TABLE db.global(k INT PRIMARY KEY, v INT) LOCALITY GLOBAL;
          ----
    datadriven_test.go:151: 
        /mnt/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/global_tables:12:
          exec-sql idx=0
          INSERT INTO db.global VALUES (1, 1), (2,2)
          ----
    datadriven_test.go:397: wait-for-zone-config-changes: initial replica check passed (5 replicas served locally)
    datadriven.go:357: /mnt/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/global_tables:16: still running after 10.002121818s
    datadriven_test.go:397: wait-for-zone-config-changes: initial replica check passed (4 replicas served locally)
    datadriven_test.go:151: 
        /mnt/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/global_tables:16:
          wait-for-zone-config-changes db-name=db table-name=global num-voters=3 num-non-voters=2 leaseholder=0 voter=1,2 non-voter=3,4 not-present=5
          ----
    datadriven_test.go:151: 
        /mnt/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/global_tables:19:
          refresh-range-descriptor-cache idx=0 table-name=global
          SELECT * FROM db.global WHERE k = 2
          ----
          LEAD_FOR_GLOBAL_READS
    datadriven_test.go:151: 
        /mnt/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/global_tables:24:
          trace-sql idx=0
          SELECT * FROM db.global WHERE k = 1
          ----
          served locally: true
          served via follower read: false
    datadriven_test.go:151: 
        /mnt/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/global_tables:30:
          refresh-range-descriptor-cache idx=1 table-name=global
          SELECT * FROM db.global WHERE k = 2
        
        expected:
          LEAD_FOR_GLOBAL_READS
        
        found:
          LAG_BY_CLUSTER_SETTING
    panic.go:615: -- test log scope end --
test logs left over in: outputs.zip/logTestMultiRegionDataDriven_global_tables1272213306
--- FAIL: TestMultiRegionDataDriven_global_tables (63.19s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=4</code>
 - <code>shard=2</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMultiRegionDataDriven_global_tables.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-62524

--- a/pkg/ccl/multiregionccl/datadriven_test.go
+++ b/pkg/ccl/multiregionccl/datadriven_test.go
@@ -359,7 +359,7 @@ SET CLUSTER SETTING kv.closed_timestamp.lead_for_global_reads_auto_tune.enabled 
 				s := ds.tc.ApplicationLayer(idx)
 				cache := s.DistSenderI().(*kvcoord.DistSender).RangeDescriptorCache()
 				tablePrefix := keys.MustAddr(s.Codec().TablePrefix(tableID))
-				entry, err := cache.TestingGetCached(ctx, tablePrefix, false, roachpb.LAG_BY_CLUSTER_SETTING)
+				entry, err := cache.TestingGetCached(ctx, tablePrefix, false, roachpb.LEAD_FOR_GLOBAL_READS)
 				if err != nil {
 					return err.Error()
 				}

--- a/pkg/ccl/multiregionccl/roundtrips_test.go
+++ b/pkg/ccl/multiregionccl/roundtrips_test.go
@@ -121,7 +121,7 @@ func TestEnsureLocalReadsOnGlobalTables(t *testing.T) {
 			// Check that the cache was indeed populated.
 			cache := tc.Server(i).DistSenderI().(*kvcoord.DistSender).RangeDescriptorCache()
 			entry, err := cache.TestingGetCached(
-				context.Background(), tablePrefix, false /* inverted */, roachpb.LAG_BY_CLUSTER_SETTING,
+				context.Background(), tablePrefix, false /* inverted */, roachpb.LEAD_FOR_GLOBAL_READS,
 			)
 			require.NoError(t, err)
 			require.False(t, entry.Lease.Empty())

